electricpy/__init__.py

```diff
@@ -6573,9 +6573,277 @@ def wireresistance(length=None,diameter=None,rho=16.8*10**-9,R=None):
         A = rho*length/R
         return _np.sqrt(4*A/pi)
         
+def ic_555(mode,R=None,C=None,f=None,t_high=None,t_low=None):
+
+    if mode == 'ASTABLE':
+
+        if R!=None and C!=None:
+
+            try:
+                assert len(R) == 2
+            except AssertionError:
+                raise ValueError(f"ic_555 in {mode} will have only 2 resitances to be fixed but {len(R)} were \
+                                given")
+
+            [R1, R2] = R
+
+            T = _np.log(2)*C*(R1+2*R2)
+            f = 1/T
+            t_low = _np.log(2)*C*R2
+            t_high = _np.log(2)*C*(R1+R2)
+            duty_cycle = t_high*100/T
+
+            return {
+                'time_period':T,
+                'frequency':f,
+                'duty_cycle':duty_cycle,
+                't_low':t_low,
+                't_high':t_high
+            }
+        
+        elif t_high!=None and t_low!=None and C!=None:
+
+            x2 = t_low/C*_np.log(2)
+            x1 = t_high/C*_np.log(2)
+
+            T = t_high+t_low
+
+            f = 1/(T)
+
+            duty_cycle = t_high/(T)
+
+            return {
+                'time_period':T,
```

Comment:
As I look at this more, it seems like this function is really doing the job of at least 3 different functions. I think it might be better, or at least more clear, if we break it out into separate functions. Perhaps some functions like:

* `ic_555_cycle_times`
* `ic_555_resistor_values`
* `ic_555_capacitor_values`

Each of those functions could still operate on the premise of allowing astable/monostable operating points, but they would be able to have a more focused usage, and that would help users.

Reply:
- ic_555_cycle_times
- ic_555_resistor_values

i have assumed that Capacitance value will be provided by user